metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -851,7 +865,16 @@ def _process_table(
                 for tag in table.column_tags[column_name]:
                     yield from self._process_tag(tag)
 
-        yield from self.gen_dataset_workunits(table, schema_name, db_name)
+            yield from self.gen_dataset_workunits(table, schema_name, db_name)
+        elif self.config.parse_view_ddl:
+            dataset_name = self.get_dataset_identifier(table.name, schema_name, db_name)
+            dataset_urn = self.gen_dataset_urn(dataset_name)
+            self.sql_parser_schema_resolver.add_schema_metadata(
+                urn=dataset_urn,
+                schema_metadata=self.get_schema_metadata(
+                    table, dataset_name, dataset_urn
+                ),
+            )
```

Comment:
You'll see this gets repeated 3 times, when it could've been written only once if put into `get_schema_metadata`. I thought that wasn't very transparent (you wouldn't think `get_schema_metadata` would cause a stateful change) so I opted to just write it 3 times

Reply:
If we renamed `get_schema_metadata` -> `gen_schema_metadata`, it might be ok to do the stateful change there

not a huge fan of the repetition here

Reply:
We can move this logic of reporting schema metadata to schema resolver and add self.config.include_schema_metadata condition inside `gen_dataset_workunits`. This way, it will be called only one.

Reply:
If keeping these multiple `get_schema_metadata` calls, we need to get this call outside `get_schema_metadata` and only in `gen_dataset_workunits`.
`        self.classify_snowflake_table(table, dataset_name, schema_metadata)`

Reply:
Alright, I'm convinced that this is not the best way